### ISSUE TITLE
Reports & statistics on errors: frontend and backend.

### ISSUE THREAD

**Issue by dwhly** [hypothesis/vision#32]
As we mature, it will be good to capture and report, and then to trigger alarms when we notice abnormal errors, like for instance "Session is invalid" that may be due to software releases, system problems, etc.  Generally we should target and address suboptimal situations based on a better understanding of their prevalence and the conditions that create them.  Good reporting infrastructure will help.

TL;DR: When this happens, we should know.
https://twitter.com/rdhyee/status/475289651591127040
